src/dex/uniswap-v2/uniswap-v2.ts

```diff
@@ -0,0 +1,909 @@
+import { AbiCoder, Interface } from '@ethersproject/abi';
+import _ from 'lodash';
+import { AsyncOrSync, DeepReadonly } from 'ts-essentials';
+import erc20ABI from '../../abi/erc20.json';
+import { StatefulEventSubscriber } from '../../stateful-event-subscriber';
+import {
+  AdapterExchangeParam,
+  Address,
+  DexConfigMap,
+  ExchangePrices,
+  Log,
+  Logger,
+  PoolLiquidity,
+  SimpleExchangeParam,
+  Token,
+  TxInfo,
+} from '../../types';
+import {
+  DexParams,
+  UniswapData,
+  UniswapDataLegacy,
+  UniswapParam,
+  UniswapPool,
+  UniswapV2Data,
+  UniswapV2Functions,
+} from './types';
+import { IDex } from '../../dex/idex';
+import {
+  ETHER_ADDRESS,
+  MAX_UINT,
+  Network,
+  NULL_ADDRESS,
+} from '../../constants';
+import { SimpleExchange } from '../simple-exchange';
+import { NumberAsString, SwapSide } from 'paraswap-core';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import {
+  wrapETH,
+  getDexKeysWithNetwork,
+  isETHAddress,
+  prependWithOx,
+} from '../../utils';
+import uniswapV2ABI from '../../abi/uniswap-v2/uniswap-v2-pool.json';
+import uniswapV2factoryABI from '../../abi/uniswap-v2/uniswap-v2-factory.json';
+import ParaSwapABI from '../../abi/IParaswap.json';
+import UniswapV2ExchangeRouterABI from '../../abi/UniswapV2ExchangeRouter.json';
+import { Contract } from '@ethersproject/contracts';
+import { WETHAddresses } from '../weth';
+import { UniswapV2Config } from './config';
+
+const Adapters: {
+  [chainId: number]: { [side: string]: { name: string; index: number }[] };
+} = {
+  [Network.MAINNET]: {
+    [SwapSide.SELL]: [
+      {
+        name: 'Adapter01',
+        index: 4,
+      },
+    ],
+    [SwapSide.BUY]: [
+      {
+        name: 'BuyAdapter',
+        index: 1,
+      },
+    ],
+  },
+  [Network.ROPSTEN]: {
+    [SwapSide.SELL]: [
+      {
+        name: 'RopstenAdapter01',
+        index: 0,
+      },
+    ],
+    [SwapSide.BUY]: [
+      {
+        name: 'RopstenBuyAdapter',
+        index: 1,
+      },
+    ],
+  },
+  [Network.POLYGON]: {
+    [SwapSide.SELL]: [
+      {
+        name: 'PolygonAdapter01',
+        index: 4,
+      },
+    ],
+    [SwapSide.BUY]: [
+      {
+        name: 'PolygonBuyAdapter',
+        index: 1,
+      },
+    ],
+  },
+  [Network.BSC]: {
+    [SwapSide.SELL]: [
+      {
+        name: 'BscAdapter01',
+        index: 3,
+      },
+    ],
```

Comment:
BakerySwap has its own adapter function since it has a different interface, can't be shared with Uniswap adapters.

Reply:
Fixed